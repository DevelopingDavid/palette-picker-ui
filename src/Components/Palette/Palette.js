import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';

const styles = theme => ({
  nested: {
    paddingLeft: theme.spacing.unit * 4,
  },
});

class Palette extends Component {

  deletePalette = async () => {
    const { id, project_id } = this.props.palette;
    const options = {method: 'DELETE'}
    await fetch(`http://localhost:3001/api/v1/projects/${project_id}/palettes/${id}`, options);
    await this.props.fetchPalette()
  }

  render() {
    const { classes } = this.props;
    const { palette } = this.props;
    return (
      <List component="div" disablePadding>
        <Divider />
        <ListItemIcon>{<i className="material-icons palette-icon">style</i>}</ListItemIcon>
        <ListItem className={classes.nested}>
          <Avatar style={{ backgroundColor: palette.color_one }}></Avatar>
          <ListItemText primary={palette.color_one} />
        </ListItem>
        <ListItem className={classes.nested}>
          <Avatar style={{ backgroundColor: palette.color_two }}></Avatar>
          <ListItemText primary={palette.color_two} />
        </ListItem>
        <ListItem className={classes.nested}>
          <Avatar style={{ backgroundColor: palette.color_three }}></Avatar>
          <ListItemText primary={palette.color_three} />
        </ListItem>
        <ListItem className={classes.nested}>
          <Avatar style={{ backgroundColor: palette.color_four }}></Avatar>
          <ListItemText primary={palette.color_four} />
        </ListItem>
        <ListItem className={classes.nested}>
          <Avatar style={{ backgroundColor: palette.color_five }}></Avatar>
          <ListItemText primary={palette.color_five} />
        </ListItem>
        <div className='delete-container'>
          <Button variant="contained" onClick={this.deletePalette}>Delete palette</Button>
        </div>
        <Divider />
      </List>
    )
  }
}

export default (withStyles(styles)(Palette));