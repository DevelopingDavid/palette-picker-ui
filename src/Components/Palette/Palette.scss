.sidebar-palette-container {
  cursor: pointer;
}