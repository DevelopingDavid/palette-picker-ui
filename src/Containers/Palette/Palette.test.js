import React from 'react';
import { Palette, mapDispatchToProps  } from './Palette';
import { createShallow } from '@material-ui/core/test-utils';

let mockPalette = {
  id: 72,
  color_one: "#00FF00",
  color_two: "#008000",
  color_three: "#00FFFF",
  color_four: "#3374FF",
  color_five: "#FF335E",
  created_at: "2019",
  updated_at: "2019",
  project_id: 51
}

let mockClasses = {
  nested: ''
}

let mockFunction = jest.fn();

describe('Palette', () => {
  let shallow;
  let wrapper
  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(
      <Palette
        palette={mockPalette}
        fetchPalette={mockFunction}
        displaySavedPalette={mockFunction}
        classes={mockClasses} />
    );
  });

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should delete palette', async () => {
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    });

    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    await wrapper.instance().deletePalette();

    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith(`http://localhost:3001/api/v1/projects/${mockPalette.project_id}/palettes/${mockPalette.id}`, {"method": "DELETE"});

  });

  it('should display call upon displaySavedPalette', () => {
    wrapper.instance().displayPalette();
    expect(mockFunction).toHaveBeenCalled();
  });

});