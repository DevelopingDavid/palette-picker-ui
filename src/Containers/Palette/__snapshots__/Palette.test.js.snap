// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match Snapshot 1`] = `ShallowWrapper {}`;

exports[`Palette should match snapshot 1`] = `undefined`;

exports[`Palette should work 1`] = `ShallowWrapper {}`;
